beacon-chain/db/kv/state.go

```diff
@@ -117,6 +118,10 @@ func (k *Store) SaveState(ctx context.Context, state *state.BeaconState, blockRo
 	}
 
 	return k.db.Update(func(tx *bolt.Tx) error {
+		if err := k.setStateSlotBitField(ctx, tx, state.Slot()); err != nil {
```

Comment:
Can this happen prior to the callback? Seems like a good idea to fail fast if this is going to fail.

Or only set this after a successful update. 

Do you need a lock too?

Reply:
Updated to set this successful after update